lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2386,12 +2386,18 @@ public function getCustomTypeDeclarationSQL(array $columnDef)
      * Obtains DBMS specific SQL code portion needed to set an index
      * declaration to be used in statements like CREATE TABLE.
      *
-     * @param array $fields
+     * @param array|Index $fields
      *
      * @return string
      */
-    public function getIndexFieldDeclarationListSQL(array $fields)
+    public function getIndexFieldDeclarationListSQL($fields)
     {
+        if ($fields instanceof Index) {
+            return implode(', ', $fields->getQuotedColumns($this));
+        } elseif (!is_array($fields)) {
```

Comment:
Don't you want to check `$fields` is of proper type before continuing? If I remove the else part, developers could call method with improper data type. Or do you just prefer to have a separate `if` statement instead of `elseif`?